### PR TITLE
Ruschein3

### PR DESCRIPTION
**Pull request by ruschein**

### CODE REVIEW

File: cpp/pipelines/ixtheo_marc_pipeline_fifo.sh [hunk 1]
```diff
@@ -137,11 +137,16 @@ mkfifo GesamtTiteldaten-post-phase"$PHASE"-"${date}".mrc
...
+StartPhase "Populate the Zeder Journal Timeliness Database Table"
+(collect_journal_stats.cc GesamtTiteldaten-post-phase"$((PHASE-1))"-"${date}".mrc >> "${log}" 2>&1 && \
+EndPhase || Abort) &
+
 
 StartPhase "Add Additional Open Access URL's"
 # Execute early for programs that rely on it for determining the open access property
 OADOI_URLS_FILE="/mnt/ZE020150/FID-Entwicklung/oadoi/oadoi_urls_ixtheo.json"
-(add_oa_urls ${OADOI_URLS_FILE} GesamtTiteldaten-post-phase"$((PHASE-1))"-"${date}".mrc \
+(add_oa_urls ${OADOI_URLS_FILE} GesamtTiteldaten-post-phase"$((PHASE-2))"-"${date}".mrc \
```
Comment (mtrojan-ub) [inline cpp/pipelines/ixtheo_marc_pipeline_fifo.sh:149]: What will happen if 2 phases read from the same FIFO without `wait`?

File: cpp/pipelines/collect_journal_stats.cc [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        system_type = "krimdok";
+    else
+        LOG_ERROR("system_type must be one of ixtheo|krimdok|relbib!");
+
+    const auto ppns_to_zeder_ids_and_types_map(GetPPNsToZederIdsAndTypesMap(system_type));
+
+    IniFile ini_file;
+    DbConnection db_connection(ini_file);
+
+    const auto marc_reader(MARC::Reader::Factory(argv[1]));
```
Comment (mtrojan-ub) [inline cpp/pipelines/collect_journal_stats.cc:209]: argv[2]?

File: cpp/pipelines/collect_journal_stats.cc [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+void ProcessRecords(MARC::Reader * const reader, const std::string &system_type,
+                    const std::unordered_map<std::string, ZederIdAndType> &ppns_to_zeder_ids_and_types_map,
+                    DbConnection * const db_connection)
+{
+    const auto zeder_instance(system_type == "krimdok" ? "krim" : system_type);
+    const auto JOB_START_TIME(std::to_string(std::time(nullptr)));
+    const auto HOSTNAME(DnsUtil::GetHostname());
+    const std::string ZEDER_URL_PREFIX("http://www-ub.ub.uni-tuebingen.de/zeder/?instanz=" + zeder_instance + "#suche=Z%3D");
```
Comment (mtrojan-ub) [inline cpp/pipelines/collect_journal_stats.cc:131]: URL is hardcoded here, Zeder::GetFullDumpEndpointPath should be used instead, as well as Zeder::Flavour

File: cpp/pipelines/collect_journal_stats.cc [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    if (__builtin_strcmp("ixtheo", argv[1]))
+        system_type = "ixtheo";
+    else if (__builtin_strcmp("krimdok", argv[1]))
+        system_type = "krimdok";
+    else
+        LOG_ERROR("system_type must be one of ixtheo|krimdok|relbib!");
+
+    const auto ppns_to_zeder_ids_and_types_map(GetPPNsToZederIdsAndTypesMap(system_type));
+
+    IniFile ini_file;
```
Comment (mtrojan-ub) [inline cpp/pipelines/collect_journal_stats.cc:206]: INI path missing (which INI file contains Zeder MySQL credentials?)

Comment (ruschein) [reply cpp/pipelines/collect_journal_stats.cc:206]: No, the path is derived from the executable name and the file resides in the standard location.

File: cpp/pipelines/collect_journal_stats.cc [hunk 5]
```diff
@@ -207,7 +208,8 @@ int Main(int argc, char *argv[]) {
     DbConnection db_connection(ini_file);
 
     const auto marc_reader(MARC::Reader::Factory(argv[1]));
-    ProcessRecords(marc_reader.get(), system_type, ppns_to_zeder_ids_and_types_map, &db_connection);
+    const auto marc_writer(MARC::Writer::Factory(argv[2]));
```
Comment (mtrojan-ub) [inline cpp/pipelines/collect_journal_stats.cc:211]: marc_reader should use argv[2] and marc_writer argv[3]

File: cpp/pipelines/ixtheo_marc_pipeline_fifo.sh [hunk 6]
```diff
@@ -138,15 +138,17 @@ mkfifo GesamtTiteldaten-post-phase"$PHASE"-"${date}".mrc
 EndPhase || Abort) &
 
 # Note: It is necessary to run this phase after articles have had their journal's PPN's inserted!
+mkfifo GesamtTiteldaten-post-phase"$PHASE"-"${date}".mrc
 StartPhase "Populate the Zeder Journal Timeliness Database Table"
 (collect_journal_stats.cc GesamtTiteldaten-post-phase"$((PHASE-1))"-"${date}".mrc >> "${log}" 2>&1 && \
+                          GesamtTiteldaten-post-phase"$PHASE"-"${date}".mrc >> "${log}" 2>&1 && \
```
Comment (mtrojan-ub) [inline cpp/pipelines/ixtheo_marc_pipeline_fifo.sh:144]: obviously wrong (e.g. duplicate >> "${log}")

**Pull request merged**